### PR TITLE
[New] Add lamejs w/ npm auto-update via single package.json

### LINKED ISSUE #12132: [Request] Add lamejs
**Library name:** lamejs
**Library description:** A Pure JavaScript MP3 Encoder
**Git repository url:** https://github.com/zhuker/lamejs
**npm package**: https://www.npmjs.com/package/lamejs
**License:** LGPL-3.0

=====================

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀

If you'll like to work locally, I'll suggest you start pulling cdnjs repository with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are so many files in this repo, without sparseCheckout, the whole process will be slow and will use more than 80GB disk space, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄

### PR DESCRIPTION
**Pull request by ParkourKarthik**
close #12132, cc @zhuker

Pull request for issue: #12132 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/zhuker/lamejs
 * Official website (optional, not the repository): 
 * NPM package url (optional): https://www.npmjs.com/package/lamejs
 * License and its reference: LGPL-3.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 14
   - Count of stars: 204
   - Count of forks: 41
 * NPM download stats (optional):
   - Downloads in the last day: 40
   - Downloads in the last week: 209
   - Downloads in the last month: 853

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by ParkourKarthik** [comment]
@sashberd Thanks for the tip. But since I'm not working offline with the repo I had to do download and fixFormat.js with dependent files and modify it to make it work.

Now I've committed the changes.

Also, I'm worried whether there should be another new pull request to be made since this pull request has more than one commit. This is my very first contribution, please tolerate my mistakes :)

**Comment by PeterDaveHello** [comment]
@ParkourKarthik no worries about that, we can squash it for you, or you'd like to learn how, we can also help ;)

**Comment by ParkourKarthik** [comment]
@PeterDaveHello Thank you 👍

**Comment by PeterDaveHello** [comment]
@ParkourKarthik you're welcome 😄

**Comment by ParkourKarthik** [comment]
@PeterDaveHello I thought squashing couldn't be done by me. As I think that the person who is authorized to merge the request would only be able to do it.

**Comment by PeterDaveHello** [comment]
@ParkourKarthik No worries, actually merging here needs more priprity, but most of our members can help squash your commits :)

**Comment by PeterDaveHello** [comment]
BTW, as this branch is not too old to failed the CI, I prefer to wait for review first, and squash later :)

File: ajax/libs/lamejs/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    "encoder",
+    "audio",
+    "webaudio"
+  ],
+  "author": "Alex Zhukov",
+  "license": "LGPL-3.0",
+  "npmName": "lamejs",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
```
Comment (sashberd) [inline ajax/libs/lamejs/package.json]: I did not see any dist folder inside any version on npm. I checked here https://libraries.io/npm/lamejs
Thus, "basePath": "" shoud be like this
and files shoud be:
"files": [
         "lame*"
       ]

File: ajax/libs/lamejs/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "lamejs",
+  "filename": "lame.min.js",
+  "description": "Pure JavaScript MP3 Encoder",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/zhuker/lamejs.git"
```
Comment (sashberd) [inline ajax/libs/lamejs/package.json]: I suggest to use  "url": "git://github.com/zhuker/lamejs.git"

Comment (sufuf3) [reply ajax/libs/lamejs/package.json]: @sashberd this is fine.
We can let this as the same with upstream https://github.com/zhuker/lamejs/blob/master/package.json#L15
Also there are some lib use this pattern, such as https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/tslib/package.json#L20

Thanks.

**Comment by ParkourKarthik** [comment]
@sashberd I'm Sorry that I had to create a new PR, since I deleted my old fork repo linked with this PR.

Fixed all the review remarks in the PR #12150

**Pull request closed without merging**